File: ArenaApp/src/main/java/msa/arena/injector/components/ApplicationComponent.java
package msa.arena.injector.components;

import android.content.Context;

import javax.inject.Named;
import javax.inject.Singleton;

import dagger.Component;
import io.reactivex.Scheduler;
import msa.arena.injector.modules.ApplicationModule;
import msa.domain.Repository;

/**
 * Created by Abhimuktheeswarar on 01-05-2017.
 */
@Singleton // Constraints this component to one-per-application or unscoped bindings.
@Component(modules = ApplicationModule.class)
public interface ApplicationComponent {

    //Exposed to sub-graphs.
    Context context();

    @Named("ui_thread")
    Scheduler uiThread();

    @Named("executor_thread")
    Scheduler executorThread();

    Repository repository();
}

File: reduxed/src/main/java/com/msa/reduxed/presentation/di/ApplicationComponent.kt
package com.msa.reduxed.presentation.di

import com.msa.reduxed.presentation.popularrepo.PopularReposActivity
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [ApplicationModule::class])
interface ApplicationComponent {

    fun inject(into: PopularReposActivity)
}
